### PR TITLE
Added Josephus Problem in C++

### PR DESCRIPTION
**Pull request by YASH01009**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Description

(Added Josephus Problem in C++)

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Issue #1525 is solved here)

### CODE REVIEW

**Comment by shraddhavp** [comment]
conflicts  to be resolved @YASH01009

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@YASH01009 Add non-recursive form too.

File: C-Plus-Plus/cp/Josephus.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+Josephus problem is a famous problem.
```
Comment (atarax665) [inline C-Plus-Plus/cp/Josephus.cpp:2]: Add problem link from any competitive programming website.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Use better name for your variables and format input output to make the code more understandable.

**Comment by YASH01009** [comment]
changes done !!

File: C-Plus-Plus/cp/Josephus.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Input : 
+4 2 
+Output :
+0 
+
+Time Complexity : O(n)
+Space Complexity : O(1)
+
+**/
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Josephus.cpp]: Add a new line at the end of the Code.

File: C-Plus-Plus/cp/Josephus.cpp (hunk 1 above)
Comment (HarshCasper) [reply C-Plus-Plus/cp/Josephus.cpp:2]: Please do add the same.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**